cli/src/cmd/cast/create2.rs

```diff
@@ -0,0 +1,274 @@
+            init_code,
+            init_code_hash,
+        } = self;
+
+        let mut regexs = vec![];
+
+        if let Some(mut matches) = matching {
+            if starts_with.is_some() || ends_with.is_some() {
+                eyre::bail!("Either use --matching or --starts/ends-with");
+            }
+
+            if matches.len() != 40 {
+                if !(matches.len() == 42 && matches.starts_with("0x")) {
+                    eyre::bail!("Please provide a 40 characters long sequence for matching");
+                }
+                matches = matches.strip_prefix("0x").unwrap().to_string();
+            }
+
+            hex::decode(&matches.replace('X', "0")).wrap_err("invalid matching hex provided")?;
+            regexs.push(matches.replace('X', "."));
+        }
+
+        if let Some(prefix) = starts_with {
+            let pad_width = prefix.len() + prefix.len() % 2;
+            hex::decode(format!("{:0>width$}", prefix, width = pad_width))
+                .wrap_err("invalid prefix hex provided")?;
+            regexs.push(format!(r"^{}", prefix));
+        }
+        if let Some(suffix) = ends_with {
+            let pad_width = suffix.len() + suffix.len() % 2;
+            hex::decode(format!("{:0>width$}", suffix, width = pad_width))
+                .wrap_err("invalid suffix hex provided")?;
+            regexs.push(format!(r"{}$", suffix));
+        }
+
+        assert!(
+            regexs.iter().map(|p| p.len() - 1).sum::<usize>() <= 40,
+            "vanity patterns length exceeded. cannot be more than 40 characters",
+        );
+
+        let regex = RegexSetBuilder::new(regexs).case_insensitive(!case_sensitive).build()?;
+
+        let init_code_hash = if let Some(init_code_hash) = init_code_hash {
+            let mut a: [u8; 32] = Default::default();
+            let init_code_hash = init_code_hash.strip_prefix("0x").unwrap_or(&init_code_hash);
+            assert!(init_code_hash.len() == 64, "init code hash should be 32 bytes long"); // 32 bytes * 2
+            a.copy_from_slice(&hex::decode(init_code_hash)?[..32]);
+            a
+        } else {
+            let init_code = init_code.strip_prefix("0x").unwrap_or(&init_code).as_bytes();
+            keccak256(hex::decode(init_code)?)
+        };
+
+        println!("Starting to generate deterministic contract address...");
+        let timer = Instant::now();
+        let (salt, addr) = std::iter::repeat(())
+            .par_bridge()
+            .map(|_| {
+                let salt = H256::random_using(&mut thread_rng());
+                let salt = Bytes::from(salt.to_fixed_bytes());
+
+                let addr = SimpleCast::to_checksum_address(&get_create2_address_from_hash(
+                    deployer,
+                    salt.clone(),
+                    init_code_hash,
+                ));
+
+                (salt, addr)
+            })
+            .find_any(move |(_, addr)| {
+                let addr = addr.to_string();
```

Comment:
is `addr` of type `Address` here?

because the `Display` impl (`.to_string`) truncates.

Reply:
I need to convert it to `String` here for the regex matching, agree with you that it truncates but it works, was the `Display` recently changed ? Logging the address on the fly doesn't shows it truncated.